cql3/restrictions/single_column_restriction.hh

```diff
@@ -384,6 +387,11 @@ public:
 class single_column_restriction::LIKE final : public single_column_restriction {
 private:
     ::shared_ptr<term> _value;
+    /// Matches cell value against LIKE pattern.  Pointer because it cannot be initialized in the
+    /// constructor when the pattern is a bind marker.  Mutable because it is initialized on demand
+    /// in is_satisfied_by().
+    mutable std::unique_ptr<like_matcher> _matcher;
```

Comment:
`std::optional` is another option here, we usually use `std::unique_ptr` when stable address is needed.